### ISSUE TITLE
Compute function and gradient simultaneously for reverse mode

### ISSUE THREAD

**Issue by unalmis** [PlasmaControl/DESC#1872]
Seems this would be useful so that the function and derivative are computed in one pass, in particular for objectives that are scalar outputs.

1. https://docs.jax.dev/en/latest/automatic-differentiation.html#evaluating-a-function-and-its-gradient-using-jax-value-and-grad
2. https://github.com/jax-ml/jax/pull/762
3. https://github.com/jax-ml/jax/pull/762#issuecomment-1002267121

**Comment by dpanici** [comment]
https://github.com/PlasmaControl/DESC/tree/dp/stoch-val-and-grad

but it gives nan right now, I had to make _x_scale static for some reason that I do not yet comprehend

**Comment by unalmis** [comment]
> FWIW, the typical optimization does have order of magnitude same fxn calls as jacobian calls. Does not remove the fact that the jac is usually (much) more expensive than f for us, but could be a savings still.

After #2170 , computing Jacobian is only 2x more time than computing the function without its derivative on my CPU. A GPU would probably crunch through the flops and prefer the ability to perform the computation in a single pass additional computation, so it might be faster to do at once on gpu.
